### PR TITLE
RSDK-7843: Do not remove resources when a remote has a transient error.

### PR DESCRIPTION
**Pull request by dgottlieb**

### CODE REVIEW

File: robot/client/client.go [hunk 1]
```diff
@@ -131,7 +131,7 @@ func isClosedPipeError(err error) bool {
 }
 
 func (rc *RobotClient) notConnectedToRemoteError() error {
-	return errors.Errorf("not connected to remote robot at %s", rc.address)
```
Comment (dgottlieb) [inline robot/client/client.go:134]: This usage adds a stack trace to the error. I found them to be an eyesore and not at all helpful.

File: robot/client/client.go [hunk 2]
```diff
@@ -346,7 +346,10 @@ func (rc *RobotClient) connectWithLock(ctx context.Context) error {
 	if err := rc.conn.Close(); err != nil {
 		return err
 	}
-	conn, err := grpc.Dial(ctx, rc.address, rc.logger, rc.dialOptions...)
+
+	cancelCtx, stop := context.WithTimeout(ctx, time.Second)
```
Comment (dgottlieb) [inline robot/client/client.go:350]: Oops, need to revert this and see if there's a better option for speeding up the test.

File: robot/client/client.go [hunk 3]
```diff
@@ -406,6 +409,7 @@ func (rc *RobotClient) checkConnection(ctx context.Context, checkEvery, reconnec
 				return
 			}
 		}
+		rc.logger.Info("Waiting to recheck. WaitTime:", waitTime)
```
Comment (dgottlieb) [inline robot/client/client.go:412]: Reverting

File: robot/client/client.go [hunk 4]
```diff
@@ -709,9 +713,8 @@ func (rc *RobotClient) PackageManager() packages.Manager {
 	return nil
 }
 
-// ResourceNames returns a list of all known resource names connected to this machine.
-//
-//	resource_names := machine.ResourceNames()
```
Comment (dgottlieb) [inline robot/client/client.go:714]: Didn't know if this served a purpose -- happy to reintroduce if there's some magic documentation thing going on here

File: robot/impl/resource_manager.go [hunk 5]
```diff
@@ -185,29 +185,22 @@ func (manager *resourceManager) updateRemoteResourceNames(
...
+	// the end has been removed and will be marked for removal.
+	for _, res := range manager.remoteResourceNames(remoteName) {
 		activeResourceNames[res] = false
 	}
 
 	anythingChanged := false
-
 	for _, resName := range newResources {
 		remoteResName := resName
-		res, err := rr.ResourceByName(remoteResName) // this returns a remote known OR foreign resource client
```
Comment (dgottlieb) [inline robot/impl/resource_manager.go:197]: Nothing changed here. Just code movement. What's important is this is now below [oldline 215/newline 208](https://github.com/viamrobotics/rdk/pull/4268/files#diff-77fd7ff51c1b7804c67c4df0a9a672ea0e3842b4a54bbd8ef81af82ac95c4f22L215). I hope the comment I added to the new code outlines the significance.

**Comment by maximpertsov** [comment]
this change is incorporated into https://github.com/viamrobotics/rdk/pull/4273, so I'm removing myself as a reviewer

**Pull request closed without merging**